src/controllers/sold.land.filter.controller.ts

```diff
@@ -13,46 +13,55 @@ import { Session } from "@companieshouse/node-session-handler";
 import { getUrlWithTransactionIdAndSubmissionId } from "../utils/url";
 
 export const get = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+
   try {
+
     logger.debugRequest(req, `GET ${config.SOLD_LAND_FILTER_PAGE}`);
 
     if (req.query[config.LANDING_PAGE_QUERY_PARAM] === '0') {
       deleteApplicationData(req.session);
     }
 
-    const appData: ApplicationData = await getApplicationData(req.session);
+    const session = req.session as Session;
+    const appData: ApplicationData = await getApplicationData(session, req);
```

Comment:
It looks a bit odd calling this method with `session` and `request` when the request contains the session. Is method overloading allowed in JS/Node and could we have a separate `getApplicationData()` function that just took a `request` and it extracted the `session`?